mlx/ops.cpp

```diff
@@ -1948,6 +1948,13 @@ array matmul(
   }
   // Type promotion
   auto out_type = promote_types(a.dtype(), b.dtype());
+  if (!is_floating_point(out_type)) {
```

Comment:
Should we throw on `&& !is_complex(out_type)` until we support `complex64_t` as well?